core/field_input.ts

```diff
@@ -582,6 +582,38 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
       );
       WidgetDiv.hideIfOwner(this);
       dropDownDiv.hideWithoutAnimation();
+    } else if (e.key === 'Tab') {
+      e.preventDefault();
+      const cursor = this.workspace_?.getCursor();
+      const target = e.shiftKey
+        ? cursor?.getPreviousNode(
+            this,
+            (node) =>
```

Comment:
Optional: these predicates are the same and could be pulled out into a helper function to make this more readable.